### PR TITLE
Missing custom formatter type in d.ts

### PR DESCRIPTION
**Pull request by GiedriusGrabauskas**
Code sample:
```ts
import * as Lint from "tslint/lib/lint";
import tslint from "gulp-tslint";

// Custom formatter class
export class Formatter extends Lint.Formatters.AbstractFormatter {
    public format(failures: Lint.RuleFailure[]): string {
        var failuresJSON = failures.map((failure: Lint.RuleFailure) => failure.toJson());
        return JSON.stringify(failuresJSON);
    }
}


gulp.task("tslint", () =>
    gulp.src("source.ts")
        .pipe(tslint({
            formatter: Formatter  // Custom formatter
        }))
        .pipe(tslint.report())
);
```

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 export interface PluginOptions {
     configuration?: any;
-    formatter?: string;
+    formatter?: string | any;
```
Comment (panuhorsmalahti) [inline index.d.ts]: Should the type be `string | Function` - or even a more strictly defined function type?

**Comment by GiedriusGrabauskas** [comment]
Changed `any` to `Function`.

**Comment by panuhorsmalahti** [comment]
Can you explain how the formatter option would work with a Function in tslint?

https://github.com/palantir/tslint/blob/master/src/lint.ts#L51

**Comment by GiedriusGrabauskas** [comment]
Waiting for response: https://github.com/palantir/tslint/issues/1427

**Comment by GiedriusGrabauskas** [comment]
Pull request (https://github.com/palantir/tslint/pull/1443) was merged.

**Comment by panuhorsmalahti** [comment]
Thanks :+1:

**Pull request merged**